components/base/Button.tsx

```diff
@@ -4,20 +4,27 @@ import { fontRobotoMono } from "@/pages/_app";
 type ButtonVariant = "primary" | "secondary";
 type ButtonProps = {
   variant?: ButtonVariant;
+  isBlock?: boolean;
 };
 
 const Button = ({
   variant = "primary",
+  isBlock = false,
   children,
   ...otherProps
 }: ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>) => {
+  const buttonBlockStyle = isBlock ? "button-blocked" : "button-unblocked";
```

Comment:
what is this use for?

Reply:
You can set true to make this Button component a block element. For example:
<img width="551" alt="image" src="https://user-images.githubusercontent.com/110534329/220007573-5f6dbf18-c094-4446-8393-073cdbabd229.png">
When you use display: flex, and flex-direction: column, your button will auto-fill to the width equal to its parent. This is a block element. For the block button, the CSS style is align-self: auto.

I use align-self to control it. Unblock button will have the CSS style: align-self: flex-start(This is the default setting). 
<img width="497" alt="image" src="https://user-images.githubusercontent.com/110534329/220007809-0828e195-be30-41e4-af4e-b93ac08ae1a8.png">